jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -182,19 +174,34 @@ def _default_flux_viewer_reference_name(self):
             self.app._jdaviz_helper, '_default_flux_viewer_reference_name', 'flux-viewer'
         )
 
+    @observe('cube_fit')
+    def _cube_fit_changed(self, msg={}):
+        if self.cube_fit:
+            self.dataset.add_filter('is_flux_cube')
+            self.dataset.remove_filter('layer_in_spectrum_viewer')
+        else:
+            self.dataset.add_filter('layer_in_spectrum_viewer')
+            self.dataset.remove_filter('is_flux_cube')
+        self.dataset._clear_cache()
+
+    @property
+    def spatial_subset(self):
+        msg = "spatial_subset is no longer supported as of 3.11 and will be removed from the user API in a future release"  # noqa
+        logging.warning(f"DeprecationWarning: {msg}")
```

Comment:
Why logger here when you throw exception in the next line anyway?

Technically there is no deprecation as you have removed that functionality by throwing an exception, so the "DeprecationWarning" is misleading.

Reply:
My intention was to deprecate access from the user API.... but since we're allowing breaking changes since this is going to be in 4.0, maybe its better to just remove them entirely and state that in the changelog.  I'll add a commit that does that and we can always revert if people prefer the gentler warning.